mllib/src/main/scala/org/apache/spark/mllib/feature/IDF.scala

```diff
@@ -218,7 +218,7 @@ private object IDFModel {
           newValues(k) = values(k) * idf(indices(k))
           k += 1
         }
-        Vectors.sparse(n, indices, newValues)
+        Vectors.sparse(n, indices, newValues).toSparse
```

Comment:
Most of mllib codebase is using

```scala
def sparse(size: Int, indices: Array[Int], values: Array[Double]): Vector =
```
which will take the raw arrays and construct the sparse vector. The `Seq` version is rarely used.